### PR TITLE
Java 17/Spring Boot 2.7 upgrades

### PR DESCRIPTION
**Pull request by danlysiak**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -81,97 +81,97 @@ def versions = [
...
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-security'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
+    implementation group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
+    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-hystrix', version: '2.1.6.RELEASE'
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.9'
+    implementation group: 'net.logstash.logback', name: 'logstash-logback-encoder', version: '5.0'
+    implementation group: 'org.apache.httpcomponents', name: 'httpclient', version: '4.5.14'
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.15.2'
+    implementation group: 'org.springdoc', name: 'springdoc-openapi-ui', version: '1.7.0'
+    implementation group: 'org.springdoc', name: 'springdoc-openapi-webmvc-core', version: '1.7.0'
```
Comment (danlysiak) [inline api/build.gradle:104]: Had to also upgrade to OpenAPI due to conflicts with other upgrades.

File: api/src/test/java/uk/gov/hmcts/payment/api/componenttests/AccountServiceTest.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public class AccountServiceTest {
     @Test
     public void retrieveExistingAccountReturnsAccountDto() throws Exception {
         String pbaCode = "PBA1234";
-        FieldSetter.setField(accountServiceImpl, accountServiceImpl.getClass().getDeclaredField("baseUrl"), baseUrl);
```
Comment (danlysiak) [inline api/src/test/java/uk/gov/hmcts/payment/api/componenttests/AccountServiceTest.java:44]: Internal method that's no longer accessible in later versions of Mockito - replaced with an equivalent.

File: api/src/test/java/uk/gov/hmcts/payment/api/componenttests/SwaggerPublisher.java [hunk 3]
```diff
@@ -63,7 +63,7 @@ public void generateDocs() throws Exception {
 
     private void generateDocsForGroup(String groupName) throws Exception {
         byte[] specs = mvc.perform(
-            get("/v2/api-docs?group=" + groupName)
+            get("/v3/api-docs?group=" + groupName)
```
Comment (danlysiak) [inline api/src/test/java/uk/gov/hmcts/payment/api/componenttests/SwaggerPublisher.java:66]: Note that the Swagger is not actually even published automatically as part of an Action - should really be fixed at some point. The current config in here doesn't quite work as expected (it generates multiple files as intended, but they all have the same content with ALL endpoints, rather than being grouped) but not related to these changes and since not published, this can probably be resolved separately?

File: api/src/main/java/uk/gov/hmcts/payment/api/configuration/security/PaymentAuthCheckerConfiguration.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
 import java.util.regex.Pattern;
 
 @Configuration
-public class AuthCheckerConfiguration {
+public class PaymentAuthCheckerConfiguration {
```
Comment (danlysiak) [inline api/src/main/java/uk/gov/hmcts/payment/api/configuration/security/PaymentAuthCheckerConfiguration.java:18]: Upgrade resulted in some additional checks on app startup that required this being named differently from the other `AuthCheckerConfiguration` from the S2S libs.

File: cve-resolution-strategy.gradle [hunk 5]
```diff
@@ -45,15 +45,6 @@ configurations.all {
                 det.useVersion '9.4.43.v20210629'
             }
 
-            /*
```
Comment (danlysiak) [inline cve-resolution-strategy.gradle:48]: General note - nearly everything deleted in here was due to compatibility issues, but should now be using later versions than what it was fixed to before anyway.

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
```
Comment (danlysiak) [inline gradle/wrapper/gradle-wrapper.properties:3]: Further changes required to go up to Gradle 8 - will probably do that under a separate PR.

**Review by davejones74 — APPROVED** [review]
